tests/phpunit/payment-methods/test-class-wc-stripe-upe-payment-gateway.php

```diff
@@ -2241,6 +2244,97 @@ public function test_pre_order_without_payment_uses_setup_intents() {
 		$this->assertTrue( (bool) $final_order->get_meta( '_stripe_upe_redirect_processed', true ) );
 	}
 
+	/**
+	 * Deposits order sets up payment intent.
+	 */
+	public function test_deposit_product_uses_setup_intents() {
```

Comment:
Thanks for adding this test. This is not a blocker for merging but it would be great if we could add tests to cover negative paths (`order_contains_deposit`) , retries (`process_order_token_payment`), and deletion logic (`delete_order_payment_token`) as well.